core/modules/replica/Configuration.cc

```diff
@@ -23,818 +23,1732 @@
+            !hostPort.insert(make_pair(completeInfo.svcHost, completeInfo.svcPort)).second ||
+            !hostPort.insert(make_pair(completeInfo.fsHost, completeInfo.fsPort)).second ||
+            !hostPort.insert(make_pair(completeInfo.dbHost, completeInfo.dbPort)).second ||
+            !hostPort.insert(make_pair(completeInfo.loaderHost, completeInfo.loaderPort)).second ||
+            !hostPort.insert(make_pair(completeInfo.exporterHost, completeInfo.exporterPort)).second ||
+            !hostPort.insert(make_pair(completeInfo.httpLoaderHost, completeInfo.httpLoaderPort)).second;
+    if (portConflictFound) {
+        LOGS(_log, LOG_LVL_ERROR, _context() << " port/host conflict in worker: " << completeInfo);
+        throw invalid_argument(
+                _context(__func__) + " port conflict detected either between the updated worker: '"
+                + completeInfo.name + "' and one of the existing workers or within the updated worker itself.");
+    }
 
+    // Update the persistent state.
+    bool const update = _data.at("workers").count(completeInfo.name) != 0;
+    if (_connectionPtr != nullptr) {
+        _connectionPtr->executeInOwnTransaction([&](decltype(_connectionPtr) conn) {
+            if (update) {
+                conn->executeSimpleUpdateQuery(
+                    "config_worker",
+                    conn->sqlEqual("name", completeInfo.name),
+                    make_pair("is_enabled", completeInfo.isEnabled),
+                    make_pair("is_read_only", completeInfo.isReadOnly),
+                    make_pair("svc_host", completeInfo.svcHost),
+                    make_pair("svc_port", completeInfo.svcPort),
+                    make_pair("fs_host", completeInfo.fsHost),
+                    make_pair("fs_port", completeInfo.fsPort),
+                    make_pair("db_host", completeInfo.dbHost),
+                    make_pair("db_port", completeInfo.dbPort),
+                    make_pair("db_user", completeInfo.dbUser),
+                    make_pair("data_dir", completeInfo.dataDir),
+                    make_pair("loader_host", completeInfo.loaderHost),
+                    make_pair("loader_port", completeInfo.loaderPort),
+                    make_pair("loader_tmp_dir", completeInfo.loaderTmpDir),
+                    make_pair("exporter_host", completeInfo.exporterHost),
+                    make_pair("exporter_port", completeInfo.exporterPort),
+                    make_pair("exporter_tmp_dir", completeInfo.exporterTmpDir),
+                    make_pair("http_loader_host", completeInfo.httpLoaderHost),
+                    make_pair("http_loader_port", completeInfo.httpLoaderPort),
+                    make_pair("http_loader_tmp_dir", completeInfo.httpLoaderTmpDir)
+                );
+            } else {
+                conn->executeInsertQuery(
+                    "config_worker",
+                    completeInfo.name,
+                    completeInfo.isEnabled,
```

Comment:
Does this depend on the order of the columns in mysql schema? How do you know that order will never change, or how do you know this order at all?

Reply:
It does depend on the order. It's highly unlikely, the order will change. And it's very likely the schema of that table will get migrated to something like:
```
(<worker-id>,<attribute-name>,<value>)
```
The current table had to be extended with additional columns on a few occasions, and each time this was causing the schema migrations.
The new schema would be easier to maintain, and it won't require schema migrations. 
So, I would rather leave this for now till the next improvement should the one be done here.